### PR TITLE
Mice can now eat cheese

### PR DESCRIPTION
**Pull request by missatessatessy**
Mice will eat any cheese they come across, and cheese can be 'fed' to them.

Fixes issue #841

(Side note, I still don't get how master stuff works, and I don't know how to keep it up to date with main yogs files without creating these prs that show up in this pr)
#### Changelog

:cl:  
tweak: Mice eat cheese
/:cl:

### CODE REVIEW

**Review by alexkar598 — CHANGES_REQUESTED** [review]
Add a cool down for the mouse eating it, a mouse is bound to go to the kitchen and just decimate your work

**Comment by Ratvar** [comment]
Add a brain to chefs so they start hiding their cheese instead of leaving it for mice to eat as well

**Comment by missatessatessy** [comment]
thats gonna be a bit harder

**Comment by Nickvr628** [comment]
If you place a cheese wedge on top of a mouse trap on the same tile, will the mouse trigger it and die?

File: code/modules/mob/living/simple_animal/friendly/mouse.dm [hunk 1]
```diff
@@ -76,7 +78,32 @@
...
+		eat_cheese()
 
+/mob/living/simple_animal/mouse/proc/eat_cheese()
+	var/obj/item/reagent_containers/food/snacks/cheesewedge/CW = locate(/obj/item/reagent_containers/food/snacks/cheesewedge) in loc
+	if(CW && full == FALSE)
+		qdel(CW)
+		say("Squeak!")
+		visible_message("<span class='warning'>[src] gobbles up the [CW].</span>")
+		full = TRUE
+		addtimer(CALLBACK(src, .proc/digestFood), 3 MINUTES)
```
Comment (alexkar598) [inline code/modules/mob/living/simple_animal/friendly/mouse.dm]:
```suggestion
		addtimer(VARSET_CALLBACK(src, full, FALSE), 3 MINUTES)
```

File: code/modules/mob/living/simple_animal/friendly/mouse.dm [hunk 2]
```diff
@@ -76,7 +78,32 @@
...
+
+/mob/living/simple_animal/mouse/proc/digestFood()
```
Comment (alexkar598) [inline code/modules/mob/living/simple_animal/friendly/mouse.dm]: Remove this proc

File: code/modules/mob/living/simple_animal/friendly/mouse.dm [hunk 3]
```diff
@@ -34,7 +35,8 @@
 	icon_state = "mouse_[body_color]"
 	icon_living = "mouse_[body_color]"
 	icon_dead = "mouse_[body_color]_dead"
-
+	if(stat == CONSCIOUS)
```
Comment (alexkar598) [inline code/modules/mob/living/simple_animal/friendly/mouse.dm]: I....what? Remove this,there is no point in eating cheese during initialization

File: code/modules/mob/living/simple_animal/friendly/mouse.dm [hunk 4]
```diff
@@ -76,7 +78,32 @@
...
+/mob/living/simple_animal/mouse/Move()
+	. = ..()
+	if(!stat)
+		eat_cheese()
 
+/mob/living/simple_animal/mouse/proc/eat_cheese()
+	var/obj/item/reagent_containers/food/snacks/cheesewedge/CW = locate(/obj/item/reagent_containers/food/snacks/cheesewedge) in loc
+	if(CW && full == FALSE)
+		qdel(CW)
+		say("Squeak!")
```
Comment (alexkar598) [inline code/modules/mob/living/simple_animal/friendly/mouse.dm]:
```suggestion
		say("Burp!")
```

File: code/modules/mob/living/simple_animal/friendly/mouse.dm [hunk 5]
```diff
@@ -76,7 +78,32 @@
...
 					visible_message("<span class='warning'>[src] chews through the [C].</span>")
+					
+/mob/living/simple_animal/mouse/Move()
+	. = ..()
+	if(!stat)
+		eat_cheese()
 
+/mob/living/simple_animal/mouse/proc/eat_cheese()
+	var/obj/item/reagent_containers/food/snacks/cheesewedge/CW = locate(/obj/item/reagent_containers/food/snacks/cheesewedge) in loc
+	if(CW && full == FALSE)
```
Comment (alexkar598) [inline code/modules/mob/living/simple_animal/friendly/mouse.dm]:
```suggestion
	if(!QDELETED(CW) && full == FALSE)
```

File: code/modules/mob/living/simple_animal/friendly/mouse.dm [hunk 6]
```diff
@@ -76,7 +78,32 @@
...
+/mob/living/simple_animal/mouse/proc/digestFood()
+	full = FALSE
+		
+/mob/living/simple_animal/mouse/attackby(obj/item/O, mob/user, params)
+	if(istype(O, /obj/item/reagent_containers/food/snacks/cheesewedge)) 
+		to_chat(user, "<span class='notice'>You feed [O] to [src].</span>")
+		visible_message("[src] squeaks happily!")
+		qdel(O)
+	else
+		..()
```
Comment (alexkar598) [inline code/modules/mob/living/simple_animal/friendly/mouse.dm]:
```suggestion
		return ..()
```

File: code/modules/mob/living/simple_animal/friendly/mouse.dm [hunk 7]
```diff
@@ -76,7 +78,32 @@
...
+					
+/mob/living/simple_animal/mouse/Move()
+	. = ..()
+	if(!stat)
+		eat_cheese()
 
+/mob/living/simple_animal/mouse/proc/eat_cheese()
+	var/obj/item/reagent_containers/food/snacks/cheesewedge/CW = locate(/obj/item/reagent_containers/food/snacks/cheesewedge) in loc
+	if(CW && full == FALSE)
+		qdel(CW)
```
Comment (alexkar598) [inline code/modules/mob/living/simple_animal/friendly/mouse.dm]: You are deleting the cheese here but later you reference it at line 92 in the message. Move this below that message

**Review by alexkar598 — CHANGES_REQUESTED** [review]

**Pull request merged**